### PR TITLE
In my own modified build of ClickHouse with some unknown changes, StrictResizeProcessor losing abandoned_chunks occasionally

### PR DESCRIPTION
**Pull request by jenrryyou**
### Changelog category (leave one):
- Not for changelog

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
In the production environment, we discovered that the StrictResizeProcessor did not correctly handle abandoned_chunks (for reasons mentioned in the code change comments), leading to occasional data loss and exceptions (Invalid status NotActive for associated output). We replicated and fixed this issue in the test environment (with approximately one error occurring every ten thousand executions). After the fix, it was also verified in the production environment that there were no similar issues. However, constructing a CI case for this fix is challenging because we have made significant modifications to the community code, making it hard to create the same execution plan with SQL.

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
What was the bug from the user's perspective?
A test is needed.

**Comment by alexey-milovidov** [comment]
Changed from "Bug Fix" to "Not for changelog".

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit 06e674cea8ba32a8d801d77686fd37cb42364c49 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/65873/06e674cea8ba32a8d801d77686fd37cb42364c49/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: src/Processors/ResizeProcessor.cpp [hunk 1]
```diff
@@ -350,10 +355,8 @@ IProcessor::Status StrictResizeProcessor::prepare(const PortNumbers & updated_in
 
         auto & waiting_output = output_ports[input_with_data.waiting_output];
 
-        if (waiting_output.status == OutputStatus::NotActive)
-            throw Exception(ErrorCodes::LOGICAL_ERROR, "Invalid status NotActive for associated output");
```
Comment (yariks5s) [inline src/Processors/ResizeProcessor.cpp:354]: What is the reason for removing this error check?

Comment (jenrryyou) [reply src/Processors/ResizeProcessor.cpp:354]: The output port which already associated with an input port may transition to a NotActive state if the output port is updated and abandoned chunks are pushed into it.

> What is the reason for removing this error check?

**Comment by jenrryyou** [comment]
hi @yariks5s，I analyzed the failed CI cases, and they seems to be unrelated to my MR. There might be due to CI unstability that requires a retry.

File: src/Processors/ResizeProcessor.cpp (hunk 1 above)
Comment (yariks5s) [reply src/Processors/ResizeProcessor.cpp:354]: I've discussed this issue with the original author of this code and we decided that these changes are raising many questions and the solution is not really clear. Could you please write a unit test regarding this issue if you have some free time?

**Pull request closed without merging**